amp/src/test/java/org/digijava/kernel/validators/activity/OnBudgetValidatorTest.java

```diff
@@ -0,0 +1,141 @@
+package org.digijava.kernel.validators.activity;
+
+import static org.hamcrest.Matchers.anything;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.emptyIterable;
+import static org.junit.Assert.assertThat;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.digijava.kernel.ampapi.endpoints.activity.ActivityErrors;
+import org.digijava.kernel.ampapi.endpoints.activity.field.APIField;
+import org.digijava.kernel.validation.ConstraintViolation;
+import org.digijava.kernel.validation.Validator;
+import org.digijava.kernel.validators.ValidatorUtil;
+import org.digijava.module.aim.dbentity.AmpActivity;
+import org.digijava.module.aim.validator.groups.Submit;
+import org.hamcrest.Matcher;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Octavian Ciubotaru
+ */
+public class OnBudgetValidatorTest {
+
+    private static APIField activityField;
+    private static HardcodedCategoryValues categoryValues;
+
+    @BeforeClass
+    public static void setUp() {
+        activityField = ValidatorUtil.getMetaData();
+        categoryValues = new HardcodedCategoryValues();
+    }
+
+    @Test
+    public void testEmptyActivity() {
+        AmpActivity activity = new AmpActivity();
+
+        Set<ConstraintViolation> violations = getConstraintViolations(activity, Submit.class);
+
+        assertThat(violations, emptyIterable());
+    }
+
+    @Test
+    public void testOffBudget() {
+        AmpActivity activity = new AmpActivity();
+        activity.setCategories(ImmutableSet.of(categoryValues.getActivityBudgets().getOffBudget()));
+
+        Set<ConstraintViolation> violations = getConstraintViolations(activity, Submit.class);
+
+        assertThat(violations, emptyIterable());
+    }
+
+    @Test
+    public void testOnBudgetNotRequiredOnDraftSave() {
+        AmpActivity activity = new AmpActivity();
+        activity.setCategories(ImmutableSet.of(categoryValues.getActivityBudgets().getOnBudget()));
+
+        Set<ConstraintViolation> violations = getConstraintViolations(activity);
+
+        assertThat(violations, emptyIterable());
+    }
+
+    @Test
+    public void testOnBudgetAllMissingFYNotRequired() {
```

Comment:
This is something I spent time when checking the ConditionalRequiredValidator that is not causing issues :)))